pkg/operator/ceph/object/topic/provisioner.go

```diff
@@ -0,0 +1,196 @@
+/*
+Copyright 2021 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package topic to manage a rook bucket notification topic.
+package topic
+
+import (
+	"context"
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/base64"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	awsrequest "github.com/aws/aws-sdk-go/aws/request"
+	awssession "github.com/aws/aws-sdk-go/aws/session"
+	awsv4signer "github.com/aws/aws-sdk-go/aws/signer/v4"
+	"github.com/aws/aws-sdk-go/service/sns"
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
+	"github.com/rook/rook/pkg/operator/ceph/object"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Provisioner struct {
+	Client      client.Client
+	Context     *clusterd.Context
+	ClusterInfo *cephclient.ClusterInfo
+	ClusterSpec *cephv1.ClusterSpec
+}
+
+func (p *Provisioner) createSession(objectStoreName string) (*awssession.Session, error) {
+	cephObjectStore, err := p.Context.RookClientset.CephV1().CephObjectStores(p.ClusterInfo.Namespace).Get(context.TODO(), objectStoreName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	objContext, err := object.NewMultisiteContext(p.Context, p.ClusterInfo, cephObjectStore)
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Debugf("Object context name: %q endpoint: %q region: %q",
+		objContext.Name, objContext.Endpoint, objContext.ZoneGroup)
+
+	// CephClusterSpec is needed for GetAdminOPSUserCredentials()
+	objContext.CephClusterSpec = *p.ClusterSpec
+	accessKey, secretKey, err := object.GetAdminOPSUserCredentials(objContext, &cephObjectStore.Spec)
+	if err != nil {
+		return nil, err
+	}
+
+	sess, err := awssession.NewSession(
+		aws.NewConfig().
+			WithRegion(objContext.ZoneGroup).
+			WithCredentials(credentials.NewStaticCredentials(accessKey, secretKey, "")).
+			WithEndpoint(objContext.Endpoint).
+			WithMaxRetries(3).
+			WithDisableSSL(!cephObjectStore.Spec.IsTLSEnabled()).
```

Comment:
if SSL is enabled for RGW, this request most likely fail. please handle it like in newS3agent()

Reply:
done here: 52e61cfa7d03879647df44df5e42ffc955d6c44b